openml/testing.py

```diff
@@ -135,17 +172,53 @@ def _cleanup_fixture(self):
         # pauses the code execution here till all tests in the 'session' is over
         yield
         # resumes from here after all collected tests are completed
+
+        # Local file deletion
         files = os.walk(directory)
         new_file_list = []
         for root, _, filenames in files:
             for filename in filenames:
                 new_file_list.append(os.path.join(root, filename))
         # filtering the files generated during this run
         new_file_list = list(set(new_file_list) - set(old_file_list))
-        print("Files to delete in local: {}".format(new_file_list))
         for file in new_file_list:
             os.remove(file)
 
+        # Test server deletion
+        openml.config.server = TestBase.test_server
+        openml.config.apikey = TestBase.apikey
+        entity_types = list(TestBase.tracker.keys())
+        # deleting 'run' first to allow other dependent entities to be deleted
```

Comment:
You could create an order that is always followed and just iterate over it. For example
```
for entity_type in ['study', 'suite', 'run', 'flow', 'task', 'dataset']:
    do
```

Reply:
I initially thought about it. However, felt that not having an independent list of hard-coded entities can better protect against future addition of entities.

Having said that, the API call functions do check against a hard-coded list of `entity_types`.